test/auto_parallel/sharding_pass_unittest.py

```diff
@@ -50,6 +50,7 @@ def apply_pass(use_sharding=False, stage=None):
 def reset_prog():
     paddle.fluid.framework.switch_main_program(paddle.static.Program())
     paddle.fluid.framework.switch_startup_program(paddle.static.Program())
+    paddle.utils.unique_name.switch()
 
 
 class TestShardingPass(unittest.TestCase):
```

Comment:
is the results exactly equal for runing dp twice?
if so, sharding should be exactly equal

Reply:
Done